### DISCUSSION TITLE
Database migration on a custom module

### DISCUSSION THREAD

**Discussion by tiggertag** [oqtane/oqtane.framework#2333] [Q&A]
I am very new to Oqtane and EntityFramework. I created a new module using template that came with Oqtane. I wanted to use code first approach and added model classes for 2 tables. Based on couple of discussions I saw here, we can't use Microsoft.EntityFrameworkCore.Tools to generate migrations. Is that correct? If so, do I need to create it manually? If so, is there a tutorial that can guide me? Is it the same case with the database first approach as well?

**Comment by JanOlsmar** [comment]
The creator produce a file YourModuleEntityBuilders.cs and an initial migration in the Server Project (You can add fields for the table there before you build the soulution) After you need to add a file  like 01000100_AddFields.cs to add fields. Look in Oqtan blogproject for example. 
This site helped me a lot https://blazorhelpwebsite.com/ViewBlogPost/4

**Reply by tiggertag** [reply]
That link is very helpful. Thank you!
